hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -435,28 +483,60 @@ public void refreshSources(String peerId) throws IOException {
    * @param src source to clear
    */
   void removeRecoveredSource(ReplicationSourceInterface src) {
-    LOG.info("Done with the recovered queue " + src.getQueueId());
+    LOG.info("Done with recovered replication queue {}", src.getQueueId());
     this.oldsources.remove(src);
     // Delete queue from storage and memory
-    deleteQueue(src.getQueueId());
+    deleteQueue(src);
     this.walsByIdRecoveredQueues.remove(src.getQueueId());
   }
 
   /**
    * Clear the metrics and related replication queue of the specified old source
-   * @param src source to clear
    */
-  void removeSource(ReplicationSourceInterface src) {
-    LOG.info("Done with the queue " + src.getQueueId());
+  public void removeSource(ReplicationSourceInterface src) {
+    LOG.info("Done with replication queue {}", src.getQueueId());
     this.sources.remove(src.getPeerId());
     // Delete queue from storage and memory
-    deleteQueue(src.getQueueId());
+    deleteQueue(src);
     this.walsById.remove(src.getQueueId());
   }
 
   /**
-   * Delete a complete queue of wals associated with a replication source
-   * @param queueId the id of replication queue to delete
+   * Add hbase:meta replication source. Called on open of hbase:meta.
+   * @see #removeHBaseMetaSource()
+   */
+  public ReplicationSourceInterface addHBaseMetaSource() throws IOException {
+    return addSource(new ReplicationPeerImpl(this.server.getConfiguration(),
```

Comment:
Could make this ReplicationPeer a static one?

Reply:
I think so. Thats a good idea.

Reply:
Changed my mind. Takes current configuration and cluster id so would need to be made on construction and why create it on all servers when only need on server carrying hbase:meta.

Reply:
Just put it in MetaReplicationSource? If we do not need meta replication source we will not touch the code?

Reply:
Maybe... If I create and add source at this location, it would make sense; i.e. if I refactor to do as you suggest (see my questions on below so I can get better idea of your suggestion)